torch/serialization.py

```diff
@@ -706,18 +789,18 @@ def persistent_load(saved_id):
                     args = pickle_module.load(f, **pickle_load_args)
                     key, storage_id, original_tensor_type = args
                     storage = deserialized_objects[storage_id]
-                    tensor_type = storage_to_tensor_type(storage)
                     ndim, = struct.unpack('<i', f.read(4))
                     # skip next 4 bytes; legacy encoding treated ndim as 8 bytes
                     f.read(4)
-                    size = struct.unpack(f'<{ndim}q', f.read(8 * ndim))
+                    numel = struct.unpack(f'<{ndim}q', f.read(8 * ndim))
                     stride = struct.unpack(f'<{ndim}q', f.read(8 * ndim))
                     storage_offset, = struct.unpack('<q', f.read(8))
-                    tensor = tensor_type().set_(storage, storage_offset, size, stride)
+                    tensor = torch.tensor([], dtype=storage.dtype).set_(
```

Comment:
nit: it would be marginally more efficient here to do torch.empty(0, dtype=storage.dtype) as it will skip parsing the Python [] to figure out what contents to put in.